include/device.h

```diff
@@ -383,9 +382,15 @@ static inline int device_set_power_state(const struct device *dev,
 					 uint32_t device_power_state,
 					 device_pm_cb cb, void *arg)
 {
-	return dev->device_pm_control(dev,
-					 DEVICE_PM_SET_POWER_STATE,
-					 &device_power_state, cb, arg);
+	if (dev->device_pm_control) {
```

Comment:
I'd like to see this as:
```
typedef int (*device_pm_control_fn)(const struct device *dev, uint32_t command,
                                    void *context, device_pm_cb cb, void *arg);

...

   device_pm_control_fn pmc = dev->device_pm_control;

   if (pmc == NULL) {
      pmc = device_pm_control_nop;
   }

   return pcm(args...);
```

to reduce code size by getting rid of the replicated call prep.  But that can be done when we clean up this API for device power management.